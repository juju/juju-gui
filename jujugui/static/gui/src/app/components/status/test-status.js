/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');
const shapeup = require('shapeup');

const Status = require('./status');

describe('Status', function() {
  let changeState;
  let model;
  let emptyEntities;
  let generatePath;
  let entities;
  const propTypes = Status.propTypes;

  const renderComponent = (options = {}) => enzyme.shallow(
    <Status
      changeState={options.changeState || changeState}
      entities={options.entities || entities}
      generatePath={options.generatePath || generatePath}
      model={shapeup.fromShape(options.model || model, propTypes.model)} />
  );

  beforeEach(() => {
    changeState = sinon.stub();
    model = {
      cloud: 'aws',
      environmentName: 'my-model',
      modelUUID: 'myuuid',
      region: 'neutral zone',
      sla: 'advanced',
      version: '2.42.47'
    };
    emptyEntities = {
      annotations: {},
      applications: {},
      machines: {},
      relations: {},
      'remote-applications': {},
      units: {}
    };
    generatePath = sinon.stub();
    entities = {
      annotations: {},
      applications: {
        django: {
          'charm-url': '~who/xenial/django-42',
          exposed: true,
          id: 'django',
          name: 'django',
          pending: false,
          status: {
            current: 'active'
          },
          'workload-version': '1.10'
        },
        haproxy: {
          'charm-url': 'haproxy-47',
          id: 'ha',
          name: 'ha',
          pending: false,
          status: {
            current: 'error'
          },
          'workload-version': ''
        },
        mysql: {
          'charm-url': 'mysql-0',
          name: 'mysql',
          pending: true,
          status: {
            current: ''
          },
          'workload-version': ''
        }
      },
      machines: {
        'm1': {
          'agent-status': {
            current: 'pending'
          },
          id: 'm1',
          'instance-id': 'machine-1',
          addresses: [{
            scope: 'public',
            value: '1.2.3.6'
          }],
          series: 'zesty'
        },
        'm2': {
          'agent-status': {
            current: 'started',
            message: 'yes, I am started'
          },
          id: 'm2',
          'instance-id': 'machine-2',
          addresses: [{
            scope: 'public',
            value: '1.2.3.7'
          }],
          series: 'trusty'
        }
      },
      relations: {
        'rel1': {
          endpoints: [{
            'application-name': 'mysql',
            relation: {
              name: 'main',
              role: 'requirer',
              interface: 'http',
              optional: false,
              limit: 1,
              scope: 'global'
            }
          }, {
            'application-name': 'mysql',
            relation: {
              name: 'website',
              role: 'provider',
              interface: 'http',
              optional: false,
              limit: 0,
              scope: 'global'
            }
          }],
          id: 'rel1'
        },
        'rel2': {
          endpoints: [{
            'application-name': 'haproxy',
            relation: {
              name: 'website',
              role: 'provider',
              interface: 'http',
              optional: false,
              limit: 1,
              scope: 'global'
            }
          }, {
            'application-name': 'wordpress',
            relation: {
              name: 'proxy',
              role: 'requirer',
              interface: 'http',
              optional: false,
              limit: 0,
              scope: 'global'
            }
          }],
          id: 'rel2'
        }
      },
      'remote-applications': {
        haproxy: {
          name: 'haproxy',
          status: {
            current: 'unknown'
          },
          url: 'local:admin/saas.haproxy'
        },
        mongo: {
          name: 'mongo',
          status: {
            current: 'corrupting data'
          },
          url: 'local:admin/my.mongo'
        }
      },
      units: {
        'django/id0': {
          'agent-status': {
            current: 'idle'
          },
          application: 'django',
          name: 'django/0',
          'machine-id': '1',
          'public-address': '1.2.3.4',
          'port-ranges': [
            {'from-port': 80, 'to-port': 80, protocol: 'tcp'},
            {'from-port': 443, 'to-port': 443, protocol: 'tcp'}
          ],
          'workload-status': {
            current: 'installing',
            message: 'these are the voyages'
          }
        },
        'django/id1': {
          'agent-status': {
            current: 'executing'
          },
          application: 'django',
          name: 'django/1',
          'machine-id': '2',
          'public-address': '1.2.3.5',
          'port-ranges': [
            {'from-port': 80, 'to-port': 88, protocol: 'udp'}
          ],
          'workload-status': {
            current: 'error',
            message: 'exterminate!'
          }
        },
        'django/id42': {
          'agent-status': {
            current: 'idle'
          },
          application: 'django',
          name: 'django/42',
          'machine-id': '2',
          'public-address': '1.2.3.6',
          // Simulate that the unit didn't open ports yet.
          'port-ranges': [],
          'workload-status': {
            current: 'installing',
            message: ''
          }
        }
      }
    };
  });

  it('renders with no data', () => {
    const model = {};
    const wrapper = renderComponent({
      entities: emptyEntities,
      model
    });
    const expected = (
      <div className="status-view__content">
        Cannot show the status: the GUI is not connected to a model.
      </div>
    );
    assert.compareJSX(wrapper.find('.status-view__content'), expected);
  });

  it('renders with no entities', () => {
    const wrapper = renderComponent({ entities: emptyEntities });
    assert.equal(wrapper.find('BasicTable').length, 1);
  });

  it('can display status by priority', () => {
    const wrapper = renderComponent();
    assert.deepEqual(
      wrapper.find('BasicTable').at(3).prop('rows')[1].classes,
      ['status-view__table-row--error']);
  });

  it('renders with entities', () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('can link to the DNS address for a unit', () => {
    const wrapper = renderComponent();
    const section = wrapper.find('BasicTable').at(3);
    const column = section.prop('rows')[0].columns[4];
    assert.equal(column.content.props.href, 'http://1.2.3.4:80');
  });

  it('can navigate to charms from the app list', () => {
    const wrapper = renderComponent();
    const section = wrapper.find('BasicTable').at(2);
    const column = section.prop('rows')[0].columns[4];
    column.content.props.onClick({preventDefault: sinon.stub()});
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(
      changeState.args[0][0], {store: 'u/who/django/xenial/42'});
  });

  it('can navigate to machines from the unit list', () => {
    const wrapper = renderComponent();
    const section = wrapper.find('BasicTable').at(3);
    const column = section.prop('rows')[0].columns[3];
    column.content.props.onClick({
      stopPropagation: sinon.stub(),
      preventDefault: sinon.stub()
    });
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(
      changeState.args[0][0], {gui: {machines: '1', status: null}});
  });

  it('can navigate to provided apps from the relation list', () => {
    const wrapper = renderComponent();
    const section = wrapper.find('BasicTable').at(5);
    const column = section.prop('rows')[0].columns[1];
    column.content.props.onClick({preventDefault: sinon.stub()});
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(changeState.args[0][0], {
      gui: {
        inspector: {
          id: 'mysql',
          activeComponent: undefined,
          unit: null,
          unitStatus: null
        }
      }
    });
  });

  it('can navigate to consumed apps from the relation list', () => {
    const wrapper = renderComponent();
    const section = wrapper.find('BasicTable').at(5);
    const column = section.prop('rows')[0].columns[2];
    column.content.props.onClick({preventDefault: sinon.stub()});
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(changeState.args[0][0], {
      gui: {
        inspector: {
          id: 'mysql',
          activeComponent: undefined,
          unit: null,
          unitStatus: null
        }
      }
    });
  });

  it('can filter by status', () => {
    const wrapper = renderComponent();
    wrapper.find('select').simulate('change', {currentTarget: {value: 'error'}});
    const instance = wrapper.instance();
    assert.equal(instance.state.statusFilter, 'error');
  });

  it('can filter by nothing', () => {
    const wrapper = renderComponent();
    wrapper.find('select').simulate('change', {currentTarget: {value: 'none'}});
    const instance = wrapper.instance();
    assert.equal(instance.state.filter, null);
  });

  it('can show a highest status notification', () => {
    const wrapper = renderComponent();
    const instance = wrapper.instance();
    instance._setHighestStatus('error');
    instance.componentDidUpdate();
    wrapper.update();
    assert.equal(
      wrapper.find('.status-view__traffic-light').prop('className').includes(
        'status-view__traffic-light--error'),
      true);
  });
});
