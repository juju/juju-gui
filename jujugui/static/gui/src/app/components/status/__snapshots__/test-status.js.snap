// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders with entities 1`] = `
<Panel
  focus={true}
  instanceName="status-view"
  visible={true}
>
  <div
    className="status-view__content"
  >
    <React.Fragment>
      <div
        key="model"
      >
        <div
          className="twelve-col no-margin-bottom"
        >
          <div
            className="eight-col"
          >
            <h2>
              my-model
              <span
                className="status-view__traffic-light status-view__traffic-light--ok"
                onClick={[Function]}
                role="button"
                tabIndex="0"
                title="Everything is OK"
              />
            </h2>
          </div>
          <div
            className="status-view__filter-label two-col"
          >
            Filter status:
          </div>
          <div
            className="status-view__filter two-col last-col"
          >
            <select
              className="status-view__filter-select"
              onChange={[Function]}
              value="none"
            >
              <option
                className="status-view__filter-option"
                key="none"
                value="none"
              >
                none
              </option>
              <option
                className="status-view__filter-option"
                key="error"
                value="error"
              >
                error
              </option>
              <option
                className="status-view__filter-option"
                key="pending"
                value="pending"
              >
                pending
              </option>
              <option
                className="status-view__filter-option"
                key="ok"
                value="ok"
              >
                ok
              </option>
            </select>
          </div>
        </div>
        <BasicTable
          headers={
            Array [
              Object {
                "columnSize": 2,
                "content": "Cloud/Region",
              },
              Object {
                "columnSize": 2,
                "content": "Version",
              },
              Object {
                "columnSize": 1,
                "content": "SLA",
              },
              Object {
                "columnSize": 2,
                "content": "Applications",
              },
              Object {
                "columnSize": 2,
                "content": "Remote applications",
              },
              Object {
                "columnSize": 1,
                "content": "Units",
              },
              Object {
                "columnSize": 1,
                "content": "Machines",
              },
              Object {
                "columnSize": 1,
                "content": "Relations",
              },
            ]
          }
          rows={
            Array [
              Object {
                "columns": Array [
                  Object {
                    "columnSize": 2,
                    "content": "aws/neutral zone",
                  },
                  Object {
                    "columnSize": 2,
                    "content": "2.42.47",
                  },
                  Object {
                    "columnSize": 1,
                    "content": "advanced",
                  },
                  Object {
                    "columnSize": 2,
                    "content": 3,
                  },
                  Object {
                    "columnSize": 2,
                    "content": 2,
                  },
                  Object {
                    "columnSize": 1,
                    "content": 3,
                  },
                  Object {
                    "columnSize": 1,
                    "content": 2,
                  },
                  Object {
                    "columnSize": 1,
                    "content": 2,
                  },
                ],
                "key": "model",
              },
            ]
          }
        />
      </div>
      <BasicTable
        filterPredicate={[Function]}
        headerClasses={
          Array [
            "status-view__table-header",
          ]
        }
        headerColumnClasses={
          Array [
            "status-view__table-header-column",
          ]
        }
        headers={
          Array [
            Object {
              "columnSize": 3,
              "content": "SAAS",
            },
            Object {
              "columnSize": 3,
              "content": "Status",
            },
            Object {
              "columnSize": 3,
              "content": "Store",
            },
            Object {
              "columnSize": 3,
              "content": "URL",
            },
          ]
        }
        key="remote-applications"
        rowClasses={
          Array [
            "status-view__table-row",
          ]
        }
        rowColumnClasses={
          Array [
            "status-view__table-column",
          ]
        }
        rows={
          Array [
            Object {
              "columns": Array [
                Object {
                  "columnSize": 3,
                  "content": undefined,
                },
                Object {
                  "columnSize": 3,
                  "content": "unknown",
                },
                Object {
                  "columnSize": 3,
                  "content": "local",
                },
                Object {
                  "columnSize": 3,
                  "content": "admin/saas.haproxy",
                },
              ],
              "extraData": "ok",
              "key": "local:admin/saas.haproxy",
            },
            Object {
              "columns": Array [
                Object {
                  "columnSize": 3,
                  "content": undefined,
                },
                Object {
                  "columnSize": 3,
                  "content": "corrupting data",
                },
                Object {
                  "columnSize": 3,
                  "content": "local",
                },
                Object {
                  "columnSize": 3,
                  "content": "admin/my.mongo",
                },
              ],
              "extraData": "ok",
              "key": "local:admin/my.mongo",
            },
          ]
        }
        sort={[Function]}
        tableClasses={
          Array [
            "status-view__table",
          ]
        }
      />
      <BasicTable
        changeState={[Function]}
        filterPredicate={[Function]}
        generatePath={[Function]}
        headerClasses={
          Array [
            "status-view__table-header",
          ]
        }
        headerColumnClasses={
          Array [
            "status-view__table-header-column",
          ]
        }
        headers={
          Array [
            Object {
              "columnSize": 2,
              "content": "Application",
            },
            Object {
              "columnSize": 2,
              "content": "Version",
            },
            Object {
              "columnSize": 2,
              "content": "Status",
            },
            Object {
              "columnSize": 1,
              "content": "Scale",
            },
            Object {
              "columnSize": 2,
              "content": "Charm",
            },
            Object {
              "columnSize": 2,
              "content": "Store",
            },
            Object {
              "columnSize": 1,
              "content": "Rev",
            },
          ]
        }
        key="applications"
        rowClasses={
          Array [
            "status-view__table-row",
          ]
        }
        rowColumnClasses={
          Array [
            "status-view__table-column",
          ]
        }
        rows={
          Array [
            Object {
              "classes": Array [
                "status-view__table-row--ok",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": undefined,
                    "id": "django",
                    "unit": null,
                    "unitStatus": null,
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/~who/xenial/django-42/icon.svg"
                    />
                    django
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": "1.10",
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--ok"
                  >
                    active
                  </span>,
                },
                Object {
                  "columnSize": 1,
                  "content": 3,
                },
                Object {
                  "columnSize": 2,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    u/who/django/xenial
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": "jujucharms",
                },
                Object {
                  "columnSize": 1,
                  "content": 42,
                },
              ],
              "extraData": "ok",
              "key": "django",
            },
            Object {
              "classes": Array [
                "status-view__table-row--error",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": undefined,
                    "id": "ha",
                    "unit": null,
                    "unitStatus": null,
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/haproxy-47/icon.svg"
                    />
                    ha
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": "",
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--error"
                  >
                    error
                  </span>,
                },
                Object {
                  "columnSize": 1,
                  "content": 0,
                },
                Object {
                  "columnSize": 2,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    haproxy
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": "jujucharms",
                },
                Object {
                  "columnSize": 1,
                  "content": 47,
                },
              ],
              "extraData": "error",
              "key": "ha",
            },
            Object {
              "classes": Array [
                "",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": undefined,
                    "id": undefined,
                    "unit": null,
                    "unitStatus": null,
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/mysql-0/icon.svg"
                    />
                    mysql
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": "",
                },
                Object {
                  "columnSize": 2,
                  "content": null,
                },
                Object {
                  "columnSize": 1,
                  "content": 0,
                },
                Object {
                  "columnSize": 2,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    mysql
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": "jujucharms",
                },
                Object {
                  "columnSize": 1,
                  "content": 0,
                },
              ],
              "extraData": "ok",
              "key": "mysql",
            },
          ]
        }
        sort={[Function]}
        tableClasses={
          Array [
            "status-view__table",
          ]
        }
      />
      <BasicTable
        changeState={[Function]}
        filterPredicate={[Function]}
        generatePath={[Function]}
        headerClasses={
          Array [
            "status-view__table-header",
          ]
        }
        headerColumnClasses={
          Array [
            "status-view__table-header-column",
          ]
        }
        headers={
          Array [
            Object {
              "columnSize": 2,
              "content": "Unit",
            },
            Object {
              "columnSize": 2,
              "content": "Workload",
            },
            Object {
              "columnSize": 2,
              "content": "Agent",
            },
            Object {
              "columnSize": 1,
              "content": "Machine",
            },
            Object {
              "columnSize": 2,
              "content": "Public address",
            },
            Object {
              "columnSize": 1,
              "content": "Ports",
            },
            Object {
              "columnSize": 2,
              "content": "Message",
            },
          ]
        }
        key="units"
        rowClasses={
          Array [
            "status-view__table-row",
          ]
        }
        rowColumnClasses={
          Array [
            "status-view__table-column",
          ]
        }
        rows={
          Array [
            Object {
              "classes": Array [
                "status-view__table-row--pending",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": "unit",
                    "id": "django",
                    "unit": "0",
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/~who/xenial/django-42/icon.svg"
                    />
                    django/0
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--pending"
                  >
                    installing
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--ok"
                  >
                    idle
                  </span>,
                },
                Object {
                  "columnSize": 1,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    1
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": <a
                    className="status-view__link"
                    href="http://1.2.3.4:80"
                    target="_blank"
                  >
                    1.2.3.4
                  </a>,
                },
                Object {
                  "columnSize": 1,
                  "content": "80/tcp, 443/tcp",
                },
                Object {
                  "columnSize": 2,
                  "content": "these are the voyages",
                },
              ],
              "extraData": "pending",
              "key": "django/0",
            },
            Object {
              "classes": Array [
                "status-view__table-row--error",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": "unit",
                    "id": "django",
                    "unit": "1",
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/~who/xenial/django-42/icon.svg"
                    />
                    django/1
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--error"
                  >
                    error
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--pending"
                  >
                    executing
                  </span>,
                },
                Object {
                  "columnSize": 1,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    2
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": <a
                    className="status-view__link"
                    href="http://1.2.3.5:80"
                    target="_blank"
                  >
                    1.2.3.5
                  </a>,
                },
                Object {
                  "columnSize": 1,
                  "content": "80-88/udp",
                },
                Object {
                  "columnSize": 2,
                  "content": "exterminate!",
                },
              ],
              "extraData": "error",
              "key": "django/1",
            },
            Object {
              "classes": Array [
                "status-view__table-row--pending",
              ],
              "clickState": Object {
                "gui": Object {
                  "inspector": Object {
                    "activeComponent": "unit",
                    "id": "django",
                    "unit": "42",
                  },
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 2,
                  "content": <span>
                    <img
                      className="status-view__icon"
                      src="v5/~who/xenial/django-42/icon.svg"
                    />
                    django/42
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--pending"
                  >
                    installing
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--ok"
                  >
                    idle
                  </span>,
                },
                Object {
                  "columnSize": 1,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    2
                  </a>,
                },
                Object {
                  "columnSize": 2,
                  "content": "1.2.3.6",
                },
                Object {
                  "columnSize": 1,
                  "content": "",
                },
                Object {
                  "columnSize": 2,
                  "content": "",
                },
              ],
              "extraData": "pending",
              "key": "django/42",
            },
          ]
        }
        sort={[Function]}
        tableClasses={
          Array [
            "status-view__table",
          ]
        }
      />
      <BasicTable
        changeState={[Function]}
        filterPredicate={[Function]}
        generatePath={[Function]}
        headerClasses={
          Array [
            "status-view__table-header",
          ]
        }
        headerColumnClasses={
          Array [
            "status-view__table-header-column",
          ]
        }
        headers={
          Array [
            Object {
              "columnSize": 1,
              "content": "Machine",
            },
            Object {
              "columnSize": 2,
              "content": "State",
            },
            Object {
              "columnSize": 2,
              "content": "DNS",
            },
            Object {
              "columnSize": 3,
              "content": "Instance ID",
            },
            Object {
              "columnSize": 1,
              "content": "Series",
            },
            Object {
              "columnSize": 3,
              "content": "Message",
            },
          ]
        }
        key="machines"
        rowClasses={
          Array [
            "status-view__table-row",
          ]
        }
        rowColumnClasses={
          Array [
            "status-view__table-column",
          ]
        }
        rows={
          Array [
            Object {
              "classes": Array [
                "status-view__table-row--pending",
              ],
              "clickState": Object {
                "gui": Object {
                  "machines": "m1",
                  "status": null,
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 1,
                  "content": "m1",
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--pending"
                  >
                    pending
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": "1.2.3.6",
                },
                Object {
                  "columnSize": 3,
                  "content": "machine-1",
                },
                Object {
                  "columnSize": 1,
                  "content": "zesty",
                },
                Object {
                  "columnSize": 3,
                  "content": undefined,
                },
              ],
              "extraData": "pending",
              "key": "m1",
            },
            Object {
              "classes": Array [
                "status-view__table-row--ok",
              ],
              "clickState": Object {
                "gui": Object {
                  "machines": "m2",
                  "status": null,
                },
              },
              "columns": Array [
                Object {
                  "columnSize": 1,
                  "content": "m2",
                },
                Object {
                  "columnSize": 2,
                  "content": <span
                    className="status-view__status--ok"
                  >
                    started
                  </span>,
                },
                Object {
                  "columnSize": 2,
                  "content": "1.2.3.7",
                },
                Object {
                  "columnSize": 3,
                  "content": "machine-2",
                },
                Object {
                  "columnSize": 1,
                  "content": "trusty",
                },
                Object {
                  "columnSize": 3,
                  "content": "yes, I am started",
                },
              ],
              "extraData": "ok",
              "key": "m2",
            },
          ]
        }
        sort={[Function]}
        tableClasses={
          Array [
            "status-view__table",
          ]
        }
      />
      <BasicTable
        filterPredicate={[Function]}
        headerClasses={
          Array [
            "status-view__table-header",
          ]
        }
        headerColumnClasses={
          Array [
            "status-view__table-header-column",
          ]
        }
        headers={
          Array [
            Object {
              "columnSize": 3,
              "content": "Relation",
            },
            Object {
              "columnSize": 3,
              "content": "Provides",
            },
            Object {
              "columnSize": 3,
              "content": "Consumes",
            },
            Object {
              "columnSize": 3,
              "content": "Type",
            },
          ]
        }
        key="relations"
        rowClasses={
          Array [
            "status-view__table-row",
          ]
        }
        rowColumnClasses={
          Array [
            "status-view__table-column",
          ]
        }
        rows={
          Array [
            Object {
              "columns": Array [
                Object {
                  "columnSize": 3,
                  "content": "website",
                },
                Object {
                  "columnSize": 3,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    <img
                      className="status-view__icon"
                      src="v5/mysql-0/icon.svg"
                    />
                    mysql
                  </a>,
                },
                Object {
                  "columnSize": 3,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    <img
                      className="status-view__icon"
                      src="v5/mysql-0/icon.svg"
                    />
                    mysql
                  </a>,
                },
                Object {
                  "columnSize": 3,
                  "content": "regular",
                },
              ],
              "key": "rel1",
            },
            Object {
              "columns": Array [
                Object {
                  "columnSize": 3,
                  "content": "website",
                },
                Object {
                  "columnSize": 3,
                  "content": <span>
                    wordpress
                  </span>,
                },
                Object {
                  "columnSize": 3,
                  "content": <a
                    className="status-view__link"
                    onClick={[Function]}
                  >
                    <img
                      className="status-view__icon"
                      src="v5/haproxy-47/icon.svg"
                    />
                    haproxy
                  </a>,
                },
                Object {
                  "columnSize": 3,
                  "content": "regular",
                },
              ],
              "key": "rel2",
            },
          ]
        }
        sort={[Function]}
        tableClasses={
          Array [
            "status-view__table",
          ]
        }
      />
    </React.Fragment>
  </div>
</Panel>
`;
